### PR TITLE
feat: Added configuration parameter that controls the amount of connection attempts to the browser

### PR DESCRIPTION
**Pull request by Malvitus**
### Additional details
Currently, there is a hardcoded limit of 62 attempts until a connection to the browser is considered fails. For my usecase, this timeout is sometimes to low and i can't change it.

I added a config value and made sure that it is used at the corresponding places when connections to browsers is attempted.

### Steps to test
1. Try to start a browser without changing the config. Everything should work just fine
2. Set connectRetryThreshold to an absurd low value (like 5). Connection should fail due to timeout

### How has the user experience changed?
In the default usecase, Experience did not change. However when users experience connection timeouts, User can adjust the config value to stabilize their tests.

### PR Tasks
- [ x] Have tests been added/updated?
- [ x] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? (https://github.com/cypress-io/cypress-documentation/pull/5070)
- [x ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

###  Other

As this is my first PR for cypress, i am unsure if I did everything right. Should something be amiss, please contact me and I will try my best to solve the issues.

### CODE REVIEW

**Review by flotwig — CHANGES_REQUESTED** [review]
@Malvitus Thanks for opening a PR.

Instead of adding a new configuration option, could you use an environment variable instead? Something like `CYPRESS_BROWSER_CONNECTION_RETRIES`. Since the vast majority of users won't need to adjust this value there's no need to put it in config.

**Comment by Malvitus** [comment]
Sure, i can change it to an environment variable. Should I open a new PR then? I guess when using an environment variable, i can circumvent a lot of parameter passing and use it directly at the places that are relevant, so I would refactor my PR anyway

**Comment by Malvitus** [comment]
@flotwig 
I completed the changes and kept them in this PR  just in case. I did not modify the changelog yet as im unsure when this change can be released. I also adjusted my Pr in the documentation repo

**Review by flotwig — APPROVED** [review]
This looks good @Malvitus. Can you clarify your use case on why you want to add this? Why is it taking more than a minute for your browser to open?

File: packages/server/lib/browsers/firefox-util.ts [hunk 1]
```diff
@@ -41,7 +43,7 @@ const getDelayMsForRetry = (i) => {
     return 500
   }
 
-  if (i < 63) {
+  if (i < maxRetries) {
```
Comment (mjhenkes) [inline packages/server/lib/browsers/firefox-util.ts]:
```suggestion
  if (i <= maxRetries) {
```
looks like we lost a retry here

Comment (Malvitus) [reply packages/server/lib/browsers/firefox-util.ts]: Good Catch!. I commited your suggestion

**Comment by Malvitus** [comment]
@flotwig 

I confess that my setup is not that straight forward. For once, my server executing the tests is not that beefy, so execution of anything is not that fast (which I don't mind). The other factor is that I have different browser versions i want to use in my Cypress tests on a network share (as I don't want them to clutter the machine). So depending on the available bandwidth, starting a browser can take longer than expected. Again this is fine for me as I don't have that many tests. Also its not the case that the browsers never start in time, just that they sometimes take a bit longer, which introduces a flakyness to my tests. So after I saw that the connection timeout is hardcoded, I thought it might be a good idea to introduce a config option there so that users can adjust the timeout up or down if needed.

**Review by mjhenkes — APPROVED** [review]

**Comment by mjhenkes** [comment]
@Malvitus could you add a changelog entry here: https://github.com/Malvitus/cypress/blob/develop/cli/CHANGELOG.md

```md
**Features:**

 - <Insert change details>. Addressed in [#25848](https://github.com/cypress-io/cypress/pull/25848).
```

**Comment by Malvitus** [comment]
@mjhenkes Done. Do you need anything else?

**Comment by mjhenkes** [comment]
@Malvitus, looks good! Thanks for the contribution. I'll work on getting this merged today.

File: cli/CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,13 @@
 <!-- See the ../guides/writing-the-cypress-changelog.md for details on writing the changelog. -->
+
+## 12.8.0
```
Comment (mjhenkes) [inline cli/CHANGELOG.md]:
```suggestion
## 12.8.0
```

File: cli/CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,12 @@
 <!-- See the ../guides/writing-the-cypress-changelog.md for details on writing the changelog. -->
+## 12.8.0
+
+_Released (PENDING)_
```
Comment (mjhenkes) [inline cli/CHANGELOG.md]:
```suggestion
_Released xx/xx/xxxx (PENDING)_
```

File: cli/CHANGELOG.md [hunk 4]
```diff
@@ -1,4 +1,12 @@
 <!-- See the ../guides/writing-the-cypress-changelog.md for details on writing the changelog. -->
+## 12.8.0
+
+_Released xx/xx/xxxx (PENDING)_
```
Comment (mjhenkes) [inline cli/CHANGELOG.md]:
```suggestion
_Released 03/14/2023 (PENDING)_
```

**Pull request merged**